bot/exts/utilities/color.py

```diff
@@ -0,0 +1,211 @@
+import colorsys
+import json
+import logging
+import re
+from io import BytesIO
+
+from PIL import Image, ImageColor
+from discord import Embed, File
+from discord.ext import commands
+from rapidfuzz import process
+
+from bot.bot import Bot
+from bot.constants import Colours
+
+
+logger = logging.getLogger(__name__)
+
+
+ERROR_MSG = """The color code {user_color} is not a possible color combination.
+\nThe range of possible values are:
+\nRGB & HSV: 0-255
+\nCMYK: 0-100%
+\nHSL: 0-360 degrees
+\nHex: #000000-#FFFFFF
+"""
+
+COLOR_LIST = "bot/resources/utilities/ryanzec_colours.json"
+with open(COLOR_LIST) as f:
+    COLOR_JSON = json.load(f)
+
+
+# define color command
+class Color(commands.Cog):
+    """User initiated command to receive color information."""
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+
+    @commands.command(aliases=["colour"])
+    async def color(self, ctx: commands.Context, mode: str, *, user_color: str) -> None:
+        """Send information on input color code or color name."""
+        logger.info(f"{mode = }")
+        logger.info(f"{user_color = }")
```

Comment:
I don't think this and the logging below is really interesting at the info level. I'd place it in debug and maybe make it a single line instead of two.